incrementals-publisher/lib/pipeline.js

```diff
@@ -0,0 +1,66 @@
+/*
+ * This module has a few functions which help process Pipeline related metadata
+ */
+const { URL } = require('url');
+
+module.exports = {
+  /*
+   * processMetadata takes in a JSON object expected to be returned by JSON API
+   * requests to Jenkins such as:
+   *  https://ci.jenkins.io/job/structs-plugin/job/PR-36/3/api/json?tree=actions[revision[hash,pullHash],remoteUrls]
+   * 
+   * @return Object containing a `hash` and `remote_url` property
+   */
+  processMetadata: (metadata) => {
+    let response = {};
+
+    metadata.actions.forEach((action) => {
+      if (action._class == 'jenkins.scm.api.SCMRevisionAction') {
+        response.hash = action.revision.hash;
+      }
+
+      if (action._class == 'hudson.plugins.git.util.BuildData') {
```

Comment:
No do not use this any more! You must port https://github.com/jglick/incrementals-downstream-publisher/commit/c278fa32a20dfacb4c68879dd08f1bbd8c3a5442.